app/Http/Controllers/Member/DashboardController.php

```diff
@@ -75,22 +76,33 @@ public function getLogChartData(Request $request): Response
         return response($data);
     }
 
-    public function getRegistrationChartData(): Response
-    {
+	public function getCompanyData(Request $request): Response
+	{
+		$data = [];
+		$companyId = $request->user()->company_id;
+		$campaigns = DB::table('campaigns')->where('company_id', '=', $companyId)->get();
 
-        $data = [
-            'registration_form' => User::whereDoesntHave('providers')->count(),
-            'google' => User::whereHas('providers', function ($query) {
-                $query->where('provider', 'google');
-            })->count(),
-            'facebook' => User::whereHas('providers', function ($query) {
-                $query->where('provider', 'facebook');
-            })->count(),
-            'twitter' => User::whereHas('providers', function ($query) {
-                $query->where('provider', 'twitter');
-            })->count(),
-        ];
+		$data['company'] = DB::table('companies')->where('id', '=', $companyId)->get()->first();
+		$data['users'] = DB::table('users')->where('company_id', '=', $companyId)->get();
+		$data['currentUserId'] = $request->user()->id;
+		$data['campaigns'] = $campaigns;
+		//$data['newsletters'] = [];
+
+		$counts = [
+			'newsletters' => 0
+		];
+
+		foreach ($data['campaigns'] as $index => $campaign) {
+			$newsletters = DB::table('newsletters')->where('campaign_id', '=', $campaign->id)->get();
```

Comment:
Please retrieve all newsletters before foreach `->where('campaign_id', 'in', [$campaignIds])`, it will lower DB queries